rom/laghos.cpp

```diff
@@ -1334,37 +1417,40 @@ int main(int argc, char *argv[])
             const double last_dt = dt;
 
             // Adaptive time step control.
-            const double dt_est = romOptions.hyperreduce ? romOper[romOptions.window]->GetTimeStepEstimateSP() : oper->GetTimeStepEstimate(*S);
-
-            //const double dt_est = oper->GetTimeStepEstimate(*S);
-            //cout << myid << ": dt_est " << dt_est << endl;
-            if (dt_est < dt)
+            if (!rom_build_database || last_step)
```

Comment:
What happens if `rom_build_database == true`, `last_step == true`, and the timestep is reset here? Are there any issues with it recomputing the final timestep with a smaller timestep size? Since it would require more than 1 step, it would no longer be the final step. This is a very rare situation, so maybe it has not been tested. If it is a complication that should be resolved in a later PR (if it becomes an issue), then there could be just an error thrown if the timestep is reduced in the `rom_build_database` case.

Reply:
Ah, I didn't think about this. it looks like we would just go to another timestep and the second to last timestep solution would be thrown away and recalculated, and there shouldn't be a problem, but this hasn't been tested yet.

Reply:
Ended up just removing the varyTimeStep, since it was didn't perform well as an error indicator anyways. Added a test for varyBasisSize.

Reply:
My question is for any `errorIndicatorType`, including `useLastLiftedSolution`. In general, my understanding is that the final timestep needs to be run, in order to evaluate the residual at the final timestep. However, the new run (using the new ROM) could possibly reduce the timestep, which means the final timestep would not be the same. It looks like the answer is that is not possible, since `!rom_build_database == false` will prevent the timestep reduction here. Is `dt` input when building a database?

Reply:
Oops. I wrote the previous comment in the wrong thread.

Reply:
Actually, I think the final `dt` cannot be prescribed or ensured to be a certain value, but the final residual can still be used.